app/controllers/admin/api/service_contracts_controller.rb

```diff
@@ -49,37 +41,29 @@ def show
     respond_with service_contract
   end
 
-  protected
+  private
 
   def account
     @account ||= current_account.buyers.find params.require(:account_id)
   end
 
   def service_contract
-    @service_contract ||= account.bought_service_contracts.find_by(id: params.require(:id))
-  end
-
-
-  def service_contract_params
-    params.permit(service_contract: [:plan_id])
-          .fetch(:service_contract).merge(plan: service_plan)
+    @service_contract ||= account.bought_service_contracts.find(params.require(:id))
   end
 
   def service_plan
-    ServicePlan.find_by(id: service_contract_plan_id)
+    @service_plan ||= ServicePlan.provided_by(current_account).find(service_contract_plan_id)
```

Comment:
This is instead of `valid_plan_and_account`. We are limiting the search scope by the current account, and if not found, return a 404.